### DISCUSSION TITLE
What are "cached tools"?

### DISCUSSION THREAD

**Discussion by dominickpastore** [community/community#25413] [Actions]
<p>I am looking at the details for the GitHub Actions virtual environments, e.g.:</p>
<ul>
<li><a href="https://github.com/actions/virtual-environments/blob/48e266a55a3c4aa19b715b299ac5729fdb6379a8/images/linux/Ubuntu1804-README.md" rel="noopener nofollow ugc">https://github.com/actions/virtual-environments/blob/48e266a55a3c4aa19b715b299ac5729fdb6379a8/images/linux/Ubuntu1804-README.md</a></li>
<li><a href="https://github.com/actions/virtual-environments/blob/48e266a55a3c4aa19b715b299ac5729fdb6379a8/images/macos/macos-10.15-Readme.md" rel="noopener nofollow ugc">https://github.com/actions/virtual-environments/blob/48e266a55a3c4aa19b715b299ac5729fdb6379a8/images/macos/macos-10.15-Readme.md</a></li>
</ul>
<p>Each one has a section called “Cached Tools.” What exactly does this mean? I assume it means they aren’t installed by default, but that they are cached somewhere for fast installation when necessary, but how do I access them?</p>
<p>Are these just the tools that are installable with actions like <code>actions/setup-python</code>, <code>actions/setup-ruby</code>, <code>actions/setup-node</code>, etc?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="dominickpastore" data-post="1" data-topic="140010">
<div class="title">
<div class="quote-controls"></div>
 dominickpastore:</div>
<blockquote>
<p>Are these just the tools that are installable with actions like <code>actions/setup-python</code> , <code>actions/setup-ruby</code> , <code>actions/setup-node</code> , etc?</p>
</blockquote>
</aside>
<p>Basically yes, though I don’t want to rule out other uses. Most of the<code>setup-*</code> actions can install non-cached versions, too, but using the cached ones is much faster.</p>
